docs/UserGuide.md

```diff
@@ -2,117 +2,213 @@
+## Usage <a name="Usage"></a>
+For the purposes of Bamboo, the terms `Tag` and `Category` are interchangeable.
+### Fields/Glossary
+1. **expense**
+    - An expense is a cost incurred in or required for something. Every entry in **Bamboo** is an expense.
+    - Bamboo treats 2 expenses as identical if they have the same **Description**, **Amount**, and **Date**.
+    - Restrictions: Bamboo cannot store duplicate expenses i.e. 2 expenses are identical according to the criterion above.  
+    - Refer to [Commands](#commands) to see how to manage expenses using Bamboo.
 
-1. **Save Load Function**
-    - Automatically saves the state of the expense book after each operation.
-    - Automatically loads previously saved data on app start-up.
+1. **description**
+    - Description of expense.
+    - Works only in complement with [add](#add), [edit](#edit), [find](#find), [sort](#sort)
+    - Prefix: `-d`
+    - Format: `-d <description>`
+    - Restrictions: Descriptions can be up to 200 characters long.
+    - Example: `-d lunch`, `-d dinner`
 
-1. **Notifications for budget limits &lt;pending&gt;**
-1. **Track progress for long-term purchases &lt;pending&gt;**
-1. **Achievement system &lt;pending&gt;**
-1. **Graphs and data views &lt;pending&gt;**
-1. **Customisation of workflow - shortcuts etc. &lt;pending&gt;**
-1. **Simulation of future spending &lt;pending&gt;**
-1. **Multiple Accounts &lt;pending&gt;**
+1. **amount**
+    - Amount of money spent in expense.
+    - **NOTE**: Amounts provided must be between 0 and 10e<sup>9</sup>    
+    - Works only in complement with [add](#add), [edit](#edit), [find](#find), [sort](#sort), [topup](#topup)
+    - Prefix: `-$`
+    - Format: `-$<dollars>[.<cents>]`
+    - Restrictions: Value provided must be between 0 and 10e<sup>9</sup>, inclusive.
+    - Example: `-$2`, `-$1.50`
 
-## Usage/ API <a name="Usage"></a>
-For the purposes of Bamboo, the terms `Tag` and `Category` are interchangeable.
-### Commands
+1. **date**
+    - Date which expense was made.
+    - Works only in complement with [add](#add), [edit](#edit), [find](#find), [sort](#sort)
+    - Prefix: `-@`
+    - Format: `-@<dd-MM-yyyy>`
+    - Restrictions: Date must be a valid calendar date.
+    - Example: `-@01-07-2020`, `-@02-07-2020`
+
+1. **tag** <a name="tag"></a>
+    - Tags expense under a category. (synonymous with Category)
+    - Each tag (or category) is associated with 1 budget, and vice versa.
+    - Works only in complement with [add](#add), [edit](#edit), [find](#find), [switch](#switch), [addCat](#addCat), [deleteCat](#deleteCat), [topup](#topup)
+    - Format: `t/<category>`
+    - Example: `edit t/Food`, `find -d Coffee t/Food`
+
+1. **remark** <a name="remark"></a>
+    - Additional information about an expense.
+    - Works only in complement with [remark](#remark)
+    - Format: `-r <remark>`
+    - Restrictions: Remarks can be up to 200 characters long.
+    - Example: `edit t/Food`, `find -d Coffee t/Food`
```

Comment:
I think this example is inaccurate. Maybe `remark 1 -r impulse spending` or `remark 2 -r emergency` might be better examples?

Reply:
Fixed as well!